source/gems/dry-validation/comparison-with-activemodel.html.md

```diff
@@ -0,0 +1,589 @@
+
+Internally, ActiveModel's `presence` validation calls the method `present?` on the validated attribute, which is equivalent to `!blank?`. Neither `present?` nor `blank?` are a inbuilt Ruby methods, but a monkey-patch added to every object by ActiveSupport, with the following semantics:
+
+- `nil` and `false` are blank
+- strings composed only of whitespace are blank
+- empty arrays and hashes are blank
+- any object that responds to `empty?` and is empty is blank.
+- everything else is present.
+
+dry-validation's `filled?` predicate is simpler than this, and considers everything to be filled except `nil`, empty Strings, empty Arrays, and empty Hashes.
+
+If you want to validate that a string key contains non-whitespace characters (like ActiveSupport's `String#present?`, you can use a custom predicate such as:
+
+```ruby
+WHITESPACE_PATTERN = /\A[[:space:]#{"\u200B\u200C\u200D\u2060\uFEFF"}]*\z/
+
+def non_blank?(input)
+  !(WHITESPACE_PATTERN =~ input)
+end
+```
+
+**Associations**
+
+If you want to be sure that an association is present, you'll need to create a custom predicate to test whether the associated object itself is present.
+
+If you want to replicate ActiveModel's presence validation of an object associated via a has_one or has_many relationship (checking `.blank?` and `.marked_for_destruction?`), you will need a custom predicate.
+
+**Booleans**
+
+If you want to validate the presence of a boolean field (e.g. true or false) you should use the built in predicate `.bool?`.
+E.g. `required(:attr).filled(:bool?)`
+
+### 2.10 absence
+
+**ActiveModel Validation**
+
+```ruby
+validates :attr, absence: true
+```
+
+**dry-validation**
+
+Dry validation includes two predicates (`empty?` and `none?`) for absence. You should use whichever is most applicable to your situation, remembering that an empty string can be turned into nil using `to_nil` coercion.
+
+```ruby
+required(:attr).filled(:none?)  # only allows nil
+required(:attr).filled(:empty?) # only empty values:  "", [], {}, or nil
+```
+
+####Associations
+
+If you want to be sure that an association is absent, you'll need create a custom predicate to test whether the associated object itself is absent.
+
+> TODO: give an example of what such a predicate could looklike
+
+**Booleans**
+
+To validate the absence of a boolean field (e.g. not true or false) you should use:
+
+```ruby
+required(:attr).filled(excluded_from?: [true, false])
+```
+
+**Booleans**
+
+To validate the absence of a boolean field (e.g. not true or false) you should use:
+
+`required(:attr).filled(:none?)`
+
+### 2.11 uniqueness
+
+Rails' `uniqueness` validation is fundamentally different from the other validations because it requires a query against a database. (Accordingly, the uniqueness validation is contained within the `activerecord` gem, while other validations are part of `activemodel`.) TODO does this mean that dry-v can't handle this case? expand
+
```

Comment:
Thanks. Can you give a code example of how to inject into the schema with `with`?